index.js

```diff
@@ -84,31 +84,32 @@ function aliases(aliases) {
 }
 
 function defaults(defaults, aliases) {
+  var out = {}
+
   if (undefined !== defaults) {
     for (var key in aliases) {
       var value = defaults[key]
       var alias = toArray(aliases[key])
 
       if (undefined !== value) {
         for (var i = 0, len = alias.length; i < len; i++) {
-          defaults[alias[i]] = value
+          out[alias[i]] = value
         }
       } else {
         for (var i = 0, len = alias.length; i < len; i++) {
           if (undefined !== (value = defaults[alias[i]])) {
-            defaults[key] = value
+            out[key] = value
 
-            for (var j = 0; j < len; j++) {
-              if (i !== j) {
-                defaults[alias[j]] = value
-              }
+            for (i = 0; i < len; i++) {
```

Comment:
@jridgewell What are we losing here? Could you provide some explanation? 😄

Reply:
This PR isn't strictly necessary, but I've always found it distasteful to modify an object that's passed in (unless that's the whole point).

Reply:
@jridgewell I know! But that's kinda the point (one of the three) of getopts to be fast just because we can.

And I know how that it looks ugly to return `defaults` at the end. I'll think about it. 🙏

Reply:
PS, this change fixes https://github.com/JorgeBucaran/getopts/pull/23/files#diff-3ca1d3e657fa03b2685f8e8d63dd0ba0R49. Mainly, you're looking for the first alias that has a default specified, and using it for all the aliases. But with the current code, you'll keep looking through the aliases to see if any later ones also have a default, and you'll use _that_ value as the default.

Reply:
Thanks for the explanation! @jridgewell 

Yep, you are totally right, I was creating times alias.length unecessary passes 😭👍